src/join-9.sol

```diff
@@ -56,7 +56,11 @@ contract GemJoin9 {
         vat = VatLike(vat_);
         ilk = ilk_;
         gem = GemLike(gem_);
-        dec = GemLike(gem_).decimals();
+
+        uint256 dec_ = GemLike(gem_).decimals();
+        require(dec_ == 18, "GemJoin9/invalid-decimals");
+        dec = dec_;
```

Comment:
This is a check it hasn't been added to other gemjoins that expect an 18 decimals token. Not opposed to it either, but breaks a bit what we have been doing as standard.

Reply:
yeah, it follows `CS-6.4` report and discussion with Tal

Reply:
it might be generalized it though if we want to follow `join7` pattern as mentioned by Tal